model/vertex_query.go

```diff
@@ -212,3 +212,54 @@ func (v *Vertex) AddProperty(client queryClient, key string, value interface{})
 
 	return v.QueryRefresh(client)
 }
+
+type EdgeWithPropsAndLabel struct {
+	Label      string
+	Id         interface{}
+	Properties []interface{}
+}
+
+func (v *Vertex) AddEdges(client queryClient, inputEdges []EdgeWithPropsAndLabel) ([]Edge, error) {
+	if client == nil {
+		return []Edge{}, gremerror.NewGrammesError("AddEdges", gremerror.ErrNilClient)
+	}
+
+	var query = newTrav().V().HasID(v.ID())
+
+	for i, edge := range inputEdges {
+		if i != 0 {
+			query = query.OutV()
+		}
+		query = query.AddE(edge.Label).To(newTrav().V().HasID(edge.Id).Raw())
+
+		if len(edge.Properties)%2 != 0 {
+			return []Edge{}, gremerror.NewGrammesError("AddEdges", gremerror.ErrOddNumberOfParameters)
+		}
+
+		if len(edge.Properties) > 0 {
+			for i := 0; i < len(edge.Properties); i += 2 {
+				query.AddStep("property", edge.Properties[i], edge.Properties[i+1])
```

Comment:
So far I've seen both `AddVertex` and this method perform this operation on properties. Should we consolidate this into a single function?